### ISSUE TITLE
Deal with custom permissions

### ISSUE THREAD

**Issue by Amoki** [limdauto/drf_openapi#49]
* DRF OpenAPI version: 0.9.7
* Python version: 3.5
* Operating System: Linux

### Description
My views use custom permission to control users access but I want to allow everyone to see the schema/doc. How do I configure drf_openapi to override views permission in schema generation ?

Thanks for this nice lib !

**Comment by limdauto** [comment]
@Amoki hey it's in the doc: https://drf-openapi.readthedocs.io/en/latest/usage.html#overwriting-default-permission. Basically 

1 - Don't add `drf_openapi.urls` to your url patterns. So don't do this:

```
urlpatterns += [url(f'{API_PREFIX}/', include('drf_openapi.urls'))]
```

2 - Instead, create an url handler of your own

```
from drf_openapi.views import get_schema_view

url('schema/$', get_schema_view(url='', title="API Documentation"), name='api_schema')
```

Let me know if that works

**Comment by Amoki** [comment]
It still doesn't work :(
It seems to behave like native DRF generator: it applies views' permissions by default and I can't find how to override these permissions

**Comment by Amoki** [comment]
Here's a example of my configuration, just in case: 
```python
# urls.py
urlpatterns = [
    url(r'^admin/', admin.site.urls),
    url(r'^doc/v(?P<version>[0-9]+\.[0-9]+)', get_schema_view(url='', title="API Documentation"), name='api_schema'),
    url(r'^cloud/(?P<cloud_id>[0-9]+)/$', org_views.CloudDetail.as_view()),
]

urlpatterns = format_suffix_patterns(urlpatterns)
```
```python
# views.py
class CloudDetail(generics.RetrieveUpdateDestroyAPIView):
    permission_classes = [permissions.IsAuthenticated, CloudPermission]
    serializer_class = CloudSerializer

```

**Comment by limdauto** [comment]
@Amoki I see what you mean. I have a solution in mind. I will make a PR this weekend. Cheers

**Comment by Amoki** [comment]
Top ! :smiley:

**Comment by limdauto** [comment]
![](https://github.trello.services/images/mini-trello-icon.png) [Fix permission customization](https://trello.com/c/3yM20pFo/9-fix-permission-customization)

**Comment by limdauto** [comment]
@Amoki https://github.com/limdauto/drf_openapi/pull/54 this has landed in 0.9.8. It will be available in PyPI in a minute. You can override the permissions by subclassing the default SchemaView class provided by drf_openapi.

**Comment by limdauto** [comment]
Relevant section in the doc: https://drf-openapi.readthedocs.io/en/latest/usage.html#customization-of-the-api-view
